pkg/api/types.go

```diff
@@ -451,6 +451,45 @@ type AgentPoolProfile struct {
 	ImageRef                     *ImageReference      `json:"imageReference,omitempty"`
 }
 
+// IsEnabledAcc returns if the AcceleratedNetworkingEnabled is explicitly enabled, or the user-provided default if non explicitly enabled
+func (a *AgentPoolProfile) IsEnabledAcc() bool {
+	if a.AcceleratedNetworkingEnabled == nil {
+		// Accelerated Networking is supported on most general purpose and compute-optimized instance sizes with 2 or more vCPUs.
+		// These supported series are: D/DSv2 and F/Fs // All the others are not supported
+		// On instances that support hyperthreading, Accelerated Networking is supported on VM instances with 4 or more vCPUs.
+		// Supported series are: D/DSv3, E/ESv3, Fsv2, and Ms/Mms.
+		if strings.Contains(a.VMSize, "Standard_A") {
+			return false
+		}
+		if strings.Contains(a.VMSize, "Standard_B") {
+			return false
+		}
+		if strings.Contains(a.VMSize, "Standard_G") {
+			return false
+		}
+		if strings.Contains(a.VMSize, "Standard_H") {
+			return false
+		}
+		if strings.Contains(a.VMSize, "Standard_L") {
+			return false
+		}
+		if strings.Contains(a.VMSize, "Standard_N") {
+			return false
+		}
+		if strings.EqualFold(a.VMSize, "Standard_D1") || strings.Contains(a.VMSize, "Standard_D1_") {
+			return false
+		}
+		if strings.EqualFold(a.VMSize, "Standard_DS1") || strings.Contains(a.VMSize, "Standard_DS1_") {
+			return false
+		}
+		if strings.EqualFold(a.VMSize, "Standard_F1") || strings.EqualFold(a.VMSize, "Standard_F1s") {
+			return false
+		}
+		return true
+	}
+	return *a.AcceleratedNetworkingEnabled
```

Comment:
If the user manually sets a.AcceleratedNetworkingEnabled to true, I think we should also check if it is supported with the VM SKU, either by checking here and overriding to false if it's not, or by adding validation for the apimodel (https://github.com/Azure/acs-engine/blob/master/pkg/api/vlabs/validate.go) to make sure that if  a.AcceleratedNetworkingEnabled is true, then the VM SKU supports accelerated networking and returning an error if it doesn't. I personally think the latter is better UX so we don't silently override.

Reply:
This is a very good point, I'll work on it